components/Reservations/ReservationsTable.tsx

```diff
@@ -0,0 +1,282 @@
+
+const fuzzyFilter: FilterFn<any> = (row, columnId, value, addMeta) => {
+    const itemRank = rankItem(row.getValue(columnId), value);
+    addMeta({ itemRank });
+    return itemRank.passed;
+};
+
+const handleDeleteReservation = async (id: number) => {
+    const isConfirmed = window.confirm("Дали сигурно сакате да ја избришете резервацијата?");
+
+    if (isConfirmed) {
+        await deleteReservation(id);
+        window.location.reload();
+    }
+};
+
+const columns = [
+    {
+        accessorKey: "date",
+        header: "Датум",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "time",
+        header: "Време",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "walkDuration",
+        header: "Времетраење на прошетка",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "name",
+        header: "Име и презиме",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "email",
+        header: "Е-пошта",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "phoneNumber",
+        header: "Телефонски број",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "city",
+        header: "Град",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "municipality",
+        header: "Општина",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "address",
+        header: "Адреса",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "dogBreed",
+        header: "Раса на куче",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "walkType",
+        header: "Тип на прошетка",
+        cell: (props) => <p>{props.getValue()}</p>
+    },
+    {
+        accessorKey: "id",
+        header: "Избриши",
+        cell: (props) => (
+            <button
+                onClick={() => handleDeleteReservation(props.getValue())}
+                className="font-semibold text-red-500"
+            >
+                X
+            </button>
+        )
+    }
+];
+
+export default function ReactTable({ activeReservations, allReservations }) {
+    const [data, setData] = useState([]);
+    const [loadAllData, setLoadAllData] = useState(false);
+    const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+    const [globalFilter, setGlobalFilter] = useState("");
+
+    useEffect(() => {
+        setData(activeReservations);
+    }, [activeReservations]);
+
+    const handleLoadingData = () => {
+        setLoadAllData(!loadAllData);
+        if (!loadAllData) {
+            setData(allReservations);
+        } else {
+            setData(activeReservations);
+        }
+    };
+
+    const table = useReactTable({
```

Comment:
What I'm missing in this table is the functionality to sort by column.
https://tanstack.com/table/latest/docs/framework/react/examples/sorting